File: Medium/Maximize Number of 1's/maximize-number-of-1s.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>

using namespace std;


// } Driver Code Ends
class Solution{
public:
    // m is maximum of number zeroes allowed to flip
    // n is size of array
    int findZeroes(int arr[], int n, int m) {
        int ans=0;
        int flip=0;
        int j=0;
        for(int i=0;i<n;i++){
            if(arr[i]==0){
                flip++;
            }
            while(flip>m){
                if(arr[j]==0){
                    flip--;
                }
                j++;
            }
            ans=max(ans,i-j+1);
        }
        return ans;
        
        
        // code here
    }  
};

//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n, i, m;
        cin >> n;
        int arr[n];
        for (i = 0; i < n; i++) {
            cin >> arr[i];
        }
        cin >> m;
        Solution ob;
        auto ans = ob.findZeroes(arr, n, m);
        cout << ans << "\n";
    }
    return 0;
}
// } Driver Code Ends

File: Medium/Maximize Number of 1's/maximize-number-of-1s.java
//{ Driver Code Starts
// Initial Template for Java

import java.io.*;
import java.util.*;

class GFG {
    // Driver code
    public static void main(String[] args) throws Exception {
        BufferedReader br =
            new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine().trim());
        while (t-- > 0) {
            int n = Integer.parseInt(br.readLine().trim());
            String[] str = br.readLine().trim().split(" ");
            int[] arr = new int[n];
            for (int i = 0; i < n; i++) {
                arr[i] = Integer.parseInt(str[i]);
            }
            int m = Integer.parseInt(br.readLine().trim());

            int ans = new Solve().findZeroes(arr, n, m);

            System.out.println(ans);
        }
    }
}
// } Driver Code Ends


// User function Template for Java

class Solve {
    // m is maximum of number zeroes allowed to flip
    int findZeroes(int arr[], int n, int m) {
        int flip=0;
        int j=0;
        int maxi=0;
        for(int i=0;i<n;i++){
            if(arr[i]==0){
                flip++;
            }
            while(flip>m){
                if(arr[j]==0){
                    flip--;
                }
                j++;
            }
            maxi=Math.max(maxi,i-j+1);
            
            
        }
        return maxi;
        // code here
    }
}
